pep-0581.rst

```diff
@@ -0,0 +1,310 @@
+Question: Will it be possible to completely automate the CLA process, so
+it does not require human intervention?
+
+Create labels for issue triage
+------------------------------
+
+In bpo, we currently have the following fields for each issues:
+
+Type: behavior, crash, compile error, resource usage, security, performance, enhancement.
+Components: 2to3, Argument Clinic, asyncio, Build, Cross-build, ctypes, ...
+Priority: release blocker, deferred blocker, critical, high, normal, low
+
+We will create the corresponding labels::
+
+   type-behavior, type-crash, type-compile error, type-resource usage, ...
+
+   components-2to3, components-argument clinic, components-asyncio, ...
+
+   priority-release blocker, priority-deferred blocker, priority-critical, ...
+
+In addition, we'll create ``needs triage`` label.
+
+Create issue template
+---------------------
+
+We will create an issue template and add the following metadata::
+
+   Type: behavior | crash | compile error | resource usage (choose one)
+   Components: 2to3 | Argument Clinic | asyncio | Build | ... (can select more than one)
+   Priority: release blocker | deferred blocker | critical | ...
+   Needs backport to: 2.7 | 3.6 | 3.7
+
+The idea is to allow the issue creator to help us triage the issue.
+The above values are pre-filled in the template. Issue creator will remove texts
+that do not apply to their issue.
+
+Based on the above metadata, bedevere-bot can apply the necessary labels to the issue.
+If issue creator did not supply the above metadata, the bot will apply ``needs triage``
+label. At that point it will require a core developer to properly label the issue.
+
+Updates to bedevere
+-------------------
+
+Bedevere-bot will need to be updated to recognize the issue metadata described above,
+and apply the proper labels.
+
+Bedevere-bot can also copy over the labels to pull requests that correspond to
+the issue.
+
+Update the devguide
+-------------------
+
+Provide explanation in the devguide about new issue workflow and triage labels.
+
+Add a button in bpo to migrate the issue to GitHub
+--------------------------------------------------
+
+This will require actual update to the bpo. But I believe this effort needed
+is much less than a complete overhaul.
+
+We will create a button in bpo, that will copy over the issue description
+and associated comments into a GitHub issue.
+
+We should not be moving all open issues to GitHub. Issues with little or no
+activity should just be closed. Issues with no decision made for years should
+just be closed.
+
+If a core developer is still interested in the issue, they can
+indicate so in the bpo issue, and later use the button to migrate it over to GitHub.
+
+Make bpo readonly
+-----------------
+
+This should be the final step. Once we start using GitHub issues, make bpo
+readonly, not shut it down.
+Do not accept new registrations. Do not allow comments or issues to be created.
```

Comment:
An alternative is to keep bpo: have bpo *and* GH.

Reply:
No. Because then we need to support two different workflows and further complicate our life.

Reply:
To minimize the burden of maintaining two issue reporting/commenting workflows, one for GitHub and one for bpo, the following constraints will be made to bpo when GitHub is used for issues:
- Disable new user registration to bpo. 
- Disable new issue creation on bpo.
- Lock comments on existing bpo issues. Additional comments would be done on GitHub with a link to the archival bpo issue.